src/components/namestable/worker-edit/worker-employement-contract-work-norm-selector.component.tsx

```diff
@@ -0,0 +1,101 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import { Grid, Input, Typography } from "@material-ui/core";
+import * as _ from "lodash";
+import React, { useCallback, useMemo, useState } from "react";
+import { TimeDrawerType } from "../../../common-models/worker-info.model";
+import { DropdownButtons } from "../../common-components/dropdown-buttons/dropdown-buttons.component";
+import { TextMaskCustom } from "../../common-components/text-mask-custom/text-mask-custom.component";
+import { WorkingTimeHelper } from "../working-time.helper";
+import { FormFieldErrorLabel } from "./form-field-error-label.component";
+import { useFormFieldStyles } from "./worker-edit.models";
+import { WorkerContractTypeDependentWorkTimeSelectorOptions } from "./worker-contract-type-dependent-worknorm-selector.component";
+
+export function WorkerEmployementContractWorkNormSelector({
+  employementTime,
+  setWorkerTime,
+  setIsFieldValid: setIsFormValid,
+}: WorkerContractTypeDependentWorkTimeSelectorOptions): JSX.Element {
+  const classes = useFormFieldStyles();
+  const [selectedTimeType, setSelectedTimeType] = useState<TimeDrawerType>(TimeDrawerType.FULL);
+
+  const handleContractTimeSelect = useCallback(
+    (timeType: TimeDrawerType) => {
+      setSelectedTimeType(timeType);
+      if (timeType !== TimeDrawerType.OTHER) {
+        setWorkerTime(toWorkNorm(timeType));
+      }
+    },
+    [setSelectedTimeType, setWorkerTime]
+  );
+
+  const contractTimeDropdownOptions = useMemo(
+    () =>
```

Comment:
(): typ?

Reply:
ide proponuje (): { dataCy: string; action: () => void; label: any }[]
nwm czy ma to sens

Reply:
Mamy zdefiniowany interfejs `ButtonData`, więc użyłem go